timelock/program/src/state/timelock_state.rs

```diff
@@ -0,0 +1,258 @@
+use super::{enums, UNINITIALIZED_VERSION};
+use arrayref::{array_mut_ref, array_ref, array_refs, mut_array_refs};
+use enums::TimelockStateStatus;
+use solana_program::{
+    program_error::ProgramError,
+    program_pack::{IsInitialized, Pack, Sealed},
+    pubkey::Pubkey,
+};
+
+/// Transaction slots allowed
+pub const TRANSACTION_SLOTS: usize = 5;
+/// How many characters are allowed in the description
+pub const DESC_SIZE: usize = 200;
+/// How many characters are allowed in the name
+pub const NAME_SIZE: usize = 32;
+///Timelock state version
+pub const TIMELOCK_STATE_VERSION: u8 = 1;
+/// Timelock state
+#[derive(Clone)]
+pub struct TimelockState {
+    /// timelock set key
+    pub timelock_set: Pubkey,
+
+    ///version
+    pub version: u8,
```

Comment:
I mean, it is the version of the struct. What else would we call it?

Reply:
You'd have a new enum which specifies the struct type, that way you don't run the risk of someone using the wrong account somewhere and potentially breaking your program.

Reply:
You mean like what you did in the Swap program, with the dynamic trait?

Reply:
That would probably be overkill. I'm thinking just an enum like:
```
pub enum TimelockStructType {
    Uninitialized, // This will be 0 in u32, so you can use this to see if it's uninitialized
    Governance, // Config
    Proposal, // Set / State
    VotingRecord, // GovernanceVotingRecord
}
```

Reply:
Oh. I was using version in the sense that we may have multiple "versions" of the struct floating around. Every time we add a new column to the struct or change something we would increment the version by 1 and then have handling for each version. This is of course using the old Pack methodology. If we switch to borsh then we would also add a "key" field which I think would be what this is - it's a single byte that tells you what kind of object it is, which would be the same no matter what version.

Reply:
This enum could serve as both, and grow out as you create new versions, e.g.
```
pub enum TimelockStructType {
    Uninitialized,
    GovernanceV1,
    ProposalV1,
    VotingRecordV1,
    GovernanceV2,
    ProposalV2,
...
}
```

Reply:
Jon, that's brilliant. I love it.

Reply:
Done